### PR TITLE
Feature/GEODE 3996

### PR DESCRIPTION
**Pull request by dschneider-pivotal**
Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

**Comment by dschneider-pivotal** [comment]
I will file a geode jira ticket for this pull request

**Comment by dschneider-pivotal** [comment]
The jira ticket is: https://issues.apache.org/jira/browse/GEODE-3996

**Comment by dschneider-pivotal** [comment]
Anil's comments are addressed by ta8d63df

File: geode-core/src/test/java/org/apache/geode/internal/cache/CacheWriterGetOldValueIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+    r.putAll(putAllMap);
+
+    assertThat(cw.getSeenEntries()).isEqualTo(this.expectedValues);
+  }
+
+  @Test
+  public void doRemoveAll() {
+    RemoveAllCacheWriter<String, String> cw;
+    Region<String, String> r = createOverflowRegion();
+    put(r, "k1", "v1");
```
Comment (agingade) [inline geode-core/src/test/java/org/apache/geode/internal/cache/CacheWriterGetOldValueIntegrationTest.java]: Can we change the name to populate region and set expected values (other than put) ...I am reading this as region.put() not realizing its a test method...

**Pull request merged**